python/lsst/afw/table/schema/schema.cc

```diff
@@ -222,15 +241,21 @@ void declareSchemaType(py::module &mod) {
     clsField.def("copyRenamed", &Field<T>::copyRenamed);
     utils::python::addOutputOp(clsField, "__str__");
     utils::python::addOutputOp(clsField, "__repr__");
+    // Pickle support
+    clsField.def("__reduce__", [clsField](Field<T> const &self) {
+        return py::make_tuple(clsField, py::make_tuple(self.getName(), self.getDoc(), self.getUnits()));
+    });
```

Comment:
Why are you not using `py::pickle` here? Please explain in the code comment, since others will be wondering when they get to this bit.

Likewise for `clsSchemaItem` below

Reply:
We were following your suggestions from last year in the slack pybind11 channel. Do you think that advice no longer applies?

https://lsstc.slack.com/archives/C31M66U9L/p1531928376000442

Reply:
The pybind11 [documentation](https://pybind11-rtdtest.readthedocs.io/en/stable/advanced.html#pickling-support) clearly says to use py::pickle to implement pickle support in C++. I thought we had concluded it was much safer to do that then use `__reduce__` in C++. (None of this is relevant to supporting pickle in Python).

Reply:
Ah. I was just going off of the discussion I linked above from slack, which implied that reduce was the better choice. I'll try to convert this to `py::pickle`.